mesonpy/_editable.py

```diff
@@ -115,6 +120,9 @@ def _proc(self, command: List[str]) -> None:
     @functools.lru_cache(maxsize=1)
     def rebuild(self) -> None:
         self._debug(f'{{cyan}}{{bold}}+ rebuilding {self._project_path}{{reset}}')
+        if self.uninstall_old:
+            if os.path.exists(self.install_dir):
+                shutil.rmtree(self.install_dir)
```

Comment:
I don't like this approach. IMO saving the list of previous installed files and manually deleting the ones that are not installed by Meson anymore would be a better way to deal with this.